docs/cookbook/rag.md

```diff
@@ -0,0 +1,257 @@
+# Retrieval Augmented Generation (RAG)
+
+When you need to provide additional information to your model that it hasn't been pretrained on, we retrieve the relevant information from an external source and provide it as context to our model.
+
+## A Barebones Retrieval Example
```

Comment:
this is the only place where we use the word barebones, consider changing to basic or changing all the basic words to barebones